### PR TITLE
Validate API - promise rejection

### PR DESCRIPTION
**Pull request by 19majkel94**
Implements #40.
@pleerock, you can refactor the name to the better name as a commit to this PR and then merge 😉

### CODE REVIEW

File: src/validation/Validator.ts [hunk 1]
```diff
@@ -47,6 +47,34 @@ export class Validator {
             return executor.stripEmptyErrors(validationErrors);
         });
     }
+
+    /**
+     * Performs validation of the given object based on decorators used in given object class.
+     */
+    restrictValidate(object: Object, options?: ValidatorOptions): Promise<void>;
```
Comment (pleerock) [inline src/validation/Validator.ts]: bad name, we need to came up with better name, need to start name with `validate*`, and something like `validateOrThrow()` or better.

File: src/validation/Validator.ts [hunk 2]
```diff
@@ -47,6 +47,34 @@ export class Validator {
...
+
+    /**
+     * Performs validation of the given object based on validation schema.
+     */
+    restrictValidate(schemaName: string, object: Object, options?: ValidatorOptions): Promise<void>;
+
+    /**
+     * Performs validation of the given object based on decorators or validation schema.
+     */
+    restrictValidate(objectOrSchemaName: Object|string, objectOrValidationOptions: Object|ValidationOptions, maybeValidatorOptions?: ValidatorOptions): Promise<void> {
```
Comment (pleerock) [inline src/validation/Validator.ts]: need to do it DRY. I think this should be easy as:

```typescript
async restrictValidate(....) {
     const validationErorrs = await this.validate(...);
     if (validationErorrs.length > 0)
            return Promise.reject(validationErorrs);

     return validationErorrs;
}
```

Comment (19majkel94) [inline src/validation/Validator.ts]: It's not that simple - when you have overload signatures, you can use only them, not the 3rd function `validate(objectOrSchemaName: Object|string, objectOrValidationOptions: Object|ValidationOptions, maybeValidatorOptions?: ValidatorOptions)`. I will move the implementation to the private core method.

Comment (pleerock) [inline src/validation/Validator.ts]: ah yeah I hate this problem

**Pull request closed without merging**